signal2html/core.py

```diff
@@ -124,14 +130,51 @@ def add_mms_attachments(db, mms, backup_dir, thread_dir):
         mms.attachments.append(a)
 
 
+def get_mms_reactions(encoded_reactions, addressbook, mid):
+    reactions = []
+    try:
+        if encoded_reactions:
+            structured_reactions = StructuredReactions.loads(encoded_reactions)
+            for structured_reaction in structured_reactions.reactions:
+                rid = addressbook.get_recipient_by_address(
+                    str(structured_reaction.who)
+                )
+                reaction = Reaction(
+                    rid=rid,
+                    what=structured_reaction.what,
+                    time_sent=dt.datetime.fromtimestamp(
+                        structured_reaction.time_sent // 1000
+                    ),
+                    time_received=dt.datetime.fromtimestamp(
+                        structured_reaction.time_received // 1000
+                    ),
+                )
+                reaction.time_sent = reaction.time_sent.replace(
+                    microsecond=(structured_reaction.time_sent % 1000) * 1000
+                )
+                reaction.time_received = reaction.time_received.replace(
+                    microsecond=(structured_reaction.time_received % 1000)
+                    * 1000
+                )
+                reactions.append(reaction)
+
+        return reactions
+    except Exception as e:
```

Comment:
Catching ``Exception`` isn't great, is there a specific exception that is raised? Also, not all of these expressions look like they could fail, so maybe it's possible to place the try/except only around the code that can unexpectedly fail?

Reply:
Indeed, I can tighten this. This is meant to protect against loads() from protobuf, in case the database would not contain correct data. There are several failure modes when injecting bad data (I've seen ValueError, TypeError and IndexError).

Reply:
Looks great now, thanks!